enterprise/frontend/src/embedding/data-picker/DataSelector/tests/DataSourceSelector.unit.spec.tsx

```diff
@@ -0,0 +1,224 @@
+import userEvent from "@testing-library/user-event";
+import fetchMock from "fetch-mock";
+
+import { setupDatabasesEndpoints } from "__support__/server-mocks";
+import { renderWithProviders, screen } from "__support__/ui";
+import { getNextId } from "__support__/utils";
+import type { Database, Table } from "metabase-types/api";
+import { createMockDatabase, createMockTable } from "metabase-types/api/mocks";
+import { createSampleDatabase } from "metabase-types/api/mocks/presets";
+import {
+  createMockSettingsState,
+  createMockState,
+} from "metabase-types/store/mocks";
+
+import { DataSourceSelector } from "../DataSelector";
+
+const DATABASES = [createSampleDatabase()];
+
+const storeInitialState = createMockState({
+  settings: createMockSettingsState({
+    "enable-nested-queries": true,
+  }),
+});
+
+const AVAILABLE_MODELS: Record<AvailableModels, ("dataset" | "table")[]> = {
+  "tables-only": ["table"],
+  "tables-and-models": ["dataset", "table"],
+};
+
+type AvailableModels = "tables-only" | "tables-and-models";
+
+interface SetupOpts {
+  databases?: Database[];
+  isJoinStep?: boolean;
+  availableModels?: AvailableModels;
+  selectedTable?: {
+    id: number | undefined;
```

Comment:
since the above uses `?`

```suggestion
    id?: number;
```

Reply:
Well, I did actually want it to be
```ts
  selectedTable?: {
    id: number;
    databaseId: number;
  };
```

But that means I couldn't set a default value to it. Since I expect people to pass this option to pass every property when passing `selectedTable`.

I fixed it another way in b781689c83dabaf7a901e513e0d4e7fa1a9131c5